### PR TITLE
fix(Script/Zone): Hellfire Peninsula - Beacons not attracting mobs.

### PR DESCRIPTION
**Pull request by avarishd**
Range is eyeballed from the source.
## Changes Proposed:
-  Attract any whelps within 40y that are alive and not in combat to the beacon, then return to homepos if alive and not in combat.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/13590
- Closes https://github.com/chromiecraft/chromiecraft/issues/4334

## SOURCE:

https://www.youtube.com/watch?v=pRfxIZpjE9E
## Tests Performed:

- tested

## How to Test the Changes:

.quest add 9391
.go xyz -769 3678 29 530
Light the beacon (and the other 2)
Mobs should "rush" to the beacon (with questionably slow speed? sniff?)
Should they aggro, they will act as normal, if not they should return to spawn position after few seconds.

## Known Issues and TODO List:

- [ ] the "running" speed seems low in comparison to the video source

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Maelthyrr** [comment]
Hi, thanks for the PR. It works well. Beacons attract mobs.

**Review by IntelligentQuantum — APPROVED** [review]

File: src/server/scripts/Outland/zone_hellfire_peninsula.cpp [hunk 1]
```diff
@@ -395,6 +382,64 @@ class npc_fel_guard_hound : public CreatureScript
...
+            {
+                me->GetCreaturesWithEntryInRange(creatureList, 40, NPC_STONESCHYE_WHELP);
+                {
+                    uint8 i = 0;
+                    for (Creature* whelp : creatureList)
+                    {
+                        if (whelp->IsAlive() && !whelp->IsInCombat() && whelp->GetMotionMaster()->GetCurrentMovementGeneratorType() != HOME_MOTION_TYPE)
+                        {
+                            float o = i * 9; i++;
+                            whelp->GetMotionMaster()->MovePoint(0, me->GetPositionX() + urand(3, 4) * cos(o) , me->GetPositionY() + urand(3, 4) * sin(o), me->GetPositionZ());
```
Comment (robens) [inline src/server/scripts/Outland/zone_hellfire_peninsula.cpp:413]: final position should be calculated with a straight line from npcPos to goPos, without random and angles involved. You could try with GetFirstCollisionPosition and move there.

Comment (avarishd) [reply src/server/scripts/Outland/zone_hellfire_peninsula.cpp:413]: I was wondering how I can achieve that, will look into it. Thanks.

**Comment by LeFroid** [comment]
Tested this patch and it works perfectly in my opinion. Thanks!

**Pull request merged**